### PR TITLE
Split continuations and data namespaces in History Trie

### PR DESCRIPTION
**Pull request by dzajkowski**
## Overview
<sup>_What this PR does, and why it's needed_</sup>
This allows Continuations and Data to live under the same name (if the SpatialMatcher chooses such a state).

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>
https://rchain.atlassian.net/browse/RCHAIN-3310

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: rholang/src/main/scala/coop/rchain/rholang/interpreter/SystemProcesses.scala [hunk 1]
```diff
@@ -165,7 +165,7 @@ object SystemProcesses {
             params                                                  <- runtimeParametersRef.get
             DeployParameters(codeHash, phloRate, userId, timestamp) = params
             _ <- produce(
-              Seq(codeHash, phloRate, userId, timestamp),
+                  Seq(codeHash, phloRate, userId, timestamp),
```
Comment (odeac) [inline rholang/src/main/scala/coop/rchain/rholang/interpreter/SystemProcesses.scala]: We should do something about automatic formatting! I know it's not your fault but I'm just pointing at the issue

File: rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/history/HistoryRepositoryImpl.scala [hunk 2]
```diff
@@ -28,16 +29,12 @@ final case class HistoryRepositoryImpl[F[_]: Sync, C, P, A, K](
 )(implicit codecC: Codec[C], codecP: Codec[P], codecA: Codec[A], codecK: Codec[K])
     extends HistoryRepository[F, C, P, A, K] {
   val joinSuffixBits                = BitVector.apply("-joins".getBytes(StandardCharsets.UTF_8))
+  val dataSuffixBits                = BitVector.apply("-data".getBytes(StandardCharsets.UTF_8))
+  val continuationSuffixBits        = BitVector.apply("-continuation".getBytes(StandardCharsets.UTF_8))
```
Comment (odeac) [inline rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/history/HistoryRepositoryImpl.scala]: do you have to call apply explicitly?

File: rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/history/HistoryRepositoryImpl.scala [hunk 3]
```diff
@@ -56,6 +53,21 @@ final case class HistoryRepositoryImpl[F[_]: Sync, C, P, A, K](
     Blake2b256Hash.create(joinChannelBits.toByteVector)
   }
 
+  private def hashContinuationsChannels(channels: Seq[C]): Blake2b256Hash = {
+    val chs = channels
+      .map(c => serializeC.encode(c))
+      .sorted(util.ordByteVector)
+    val channelsBits    = codecSeq(codecByteVector).encode(chs).get
+    val joinChannelBits = channelsBits ++ continuationSuffixBits
+    Blake2b256Hash.create(joinChannelBits.toByteVector)
```
Comment (odeac) [inline rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/history/HistoryRepositoryImpl.scala]: this logic of creating a continuation channel appears almost identical in at least 3 places

File: rspace/src/test/scala/coop/rchain/rspace/nextgenrspace/history/HistoryRepositorySpec.scala [hunk 4]
```diff
@@ -48,6 +48,27 @@ class HistoryRepositorySpec
...
+      val continuations =
+        InsertContinuations[String, String, String](channel :: Nil, testContinuation :: Nil)
+      for {
+        nextRepo            <- repo.checkpoint(data :: joins :: continuations :: Nil)
+        fetchedData         <- nextRepo.getData(channel)
+        fetchedContinuation <- nextRepo.getContinuations(channel :: Nil)
+        fetchedJoins        <- nextRepo.getJoins(channel)
+        _                   = fetchedData should not be empty
+        _                   = fetchedContinuation should not be empty
+        _                   = fetchedJoins should not be empty
```
Comment (odeac) [inline rspace/src/test/scala/coop/rchain/rspace/nextgenrspace/history/HistoryRepositorySpec.scala]: a better test is if we check for the value of data, continuation & join

**Comment by dzajkowski** [comment]
bors try

File: rspace/src/test/scala/coop/rchain/rspace/nextgenrspace/history/HistoryRepositoryGenerativeSpec.scala [hunk 5]
```diff
@@ -38,6 +38,8 @@ class LMDBHistoryRepositoryGenerativeSpec
 
   val dbDir: Path = Files.createTempDirectory("rchain-storage-test-")
 
+  println("???" + dbDir)
```
Comment (ArturGajowy) [inline rspace/src/test/scala/coop/rchain/rspace/nextgenrspace/history/HistoryRepositoryGenerativeSpec.scala]: :D

**Pull request merged**